scripts/create_role_molecule.py

```diff
@@ -0,0 +1,126 @@
+#!/usr/bin/env python3
+
+from pathlib import Path
+import yaml
+
+PROJECT_DIR = Path(__file__).resolve().parent.parent
+SCRIPT_DIR = PROJECT_DIR / 'scripts'
+CI_TEMPLATES_DIR = PROJECT_DIR / 'ci' / 'templates'
+CI_CONFIG_DIR = PROJECT_DIR / 'ci' / 'config'
+ROLES_DIR = PROJECT_DIR / 'ci_framework' / 'roles'
+ZUUL_JOB_DIR = PROJECT_DIR / 'zuul.d'
+SOURCE_ROLES_DOCS_DIR = PROJECT_DIR / 'docs' / 'source' / 'roles'
+
+
+def generate_zuul_job_role_yaml(role_name):
+    """
+    Generate YAML structure for a Zuul job based on given role.
+
+    Args:
+        role_name (str): Name of the role.
+
+    Returns:
+        dict: YAML structure for the role.
+    """
+    zuul_job_base_yaml = {
```

Comment:
Why not putting this on a file? It will be easier to maintain. Maybe this can be a jinja template... That way will be, get the jinja2 template, from that create the base yaml, override the values from the yaml with your current yaml merge logic and append it to the resulting molecule.yml

Reply:
Changed into a template.